README.md

```diff
@@ -6,19 +6,109 @@ __WIP__
 - docker
 This repo is tested with python 3. 
 
-## Running Testsuite
-Start any script from `scripts/ci`.
+### Starting an Environment
 
-Tests should always be run within a docker container, as services are connected via a shared network. 
-The setup of the services needed is done by the script `scripts/start_services.py`, which is triggered from all test scripts within `scripts/ci`.
+`scripts/compose.py` provides a handy cli for starting a testing environment using docker-compose.
+`make venv` creates a virtual environment with all of the python-based dependencies needed to run `scripts/compose.py` - it requires `virtualenv` in your `PATH`.
+Activate the virtualenv with `source venv/bin/activate` and use `scripts/compose.py --help` for information on subcommands and arguments.
 
-### TODOs:
-- add tests
-- improve docker setup
-  - use ENTRYPOPINT over CMD
-  - check docker user permissions
+### Stopping
+
+All services:
+```
+compose.py stop
+
+# OR
+
+docker-compose down
+```
+
+All services and clean up volumes and networks:
+```
+make stop-env
+```
+
+Individual services:
+```
+docker-compose stop <service name>
+```
+
+#### Example environments
+
+##### Change default ports
+
+Expose Kibana on http://localhost:1234:
+
+```
+./scripts/compose.py start master --kibana-port 1234
+```
+
+##### Opbeans
+
+    Opbeans are demo web applications that are instrumented with Elastic APM.
+
+    - `./srcipts/compose.py start --all master` - start `opbeans-*` services and their dependencies along with apm-server, elasticsearch, and kibana
+
+##### Kafka output
+
+    ./scripts/compose.py start --with-kafka --with-zookeeper --apm-server-output=kafka --with-logstash master
+
+Logstash will be configured to ingest events from kafka.
+
+Topics are named according to service name. To view events for 1234_service-12a3:
+
+    docker exec -it localtesting_6.3.0-SNAPSHOT_kafka kafka-console-consumer --bootstrap-server kafka:9092 --topic apm-1234_service-12a3 --from-beginning --max-messages 100
+
+#### Advanced topics
+
+##### Dumping docker-compose.yml
+
+    `start --docker-compose-path --skip-download -` will dump the generated `docker-compose.yml` to standard out without starting any containers or downloading images.  Omit `--skip-download` to just download images.
+
+##### Testing compose
+
+    `compose.py` includes unittests, `make test-compose` to run.
+
+Onboarding events will go to the apm topic.
+
+### Running Tests
+
+All integration tests are written in python and live under `tests/`.
+
+Several `make` targets exist to make their execution simpler:
+
+- test-server
+- test-kibana
+- test-agent-{go,node,python,ruby,...}
```

Comment:
this still doesn't work for me, i only made work it once and never again :(

the issue is always a timeout on querying ES 

```
retry.py                   200 DEBUG    Converted retries value: False -> Retry(total=False, connect=None, read=None, redirect=0, status=None)
connectionpool.py          396 DEBUG    http://localhost:9200 "GET /apm-*/_count HTTP/1.1" 200 71
base.py                     83 INFO     GET http://localhost:9200/apm-*/_count [status:200 request:0.003s]
base.py                     85 DEBUG    > {"query": {"bool": {"must": [{"term": {"processor.event": {"value": "transaction"}}}]}}}
base.py                     86 DEBUG    < {"count":0,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0}}
```

will have to spend some time tracking it down but would appreciate some help to see what i'm doing wrong

Reply:
there's an issue with the current snapshots - let's rerun when something newer than jun-1 is available.